src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -170,7 +170,7 @@ module.exports = function<T, P, I, TI, PI, C, CX>(config : HostConfig<T, P, I, T
 
     deferredUpdates,
 
-    getPublicRootInstance(container : OpaqueNode) : (ReactComponent<any, any, any> | I | TI | null) {
+    getPublicRootInstance(container : OpaqueNode) : (ReactComponent<any, any, any> | PI | null) {
```

Comment:
I think this is “current” behavior. Even if I remove the `getPublicInstance` added from the ref, the result in that example is not '5' as I expected. I think, though, that that is a different issue..